numpy/_core/meson.build

```
@@ -128,6 +128,21 @@ if use_intel_sort and not fs.exists('src/npysort/x86-simd-sort/README.md')
   error('Missing the `x86-simd-sort` git submodule! Run `git submodule update --init` to fix this.')
 endif
 
+use_openmp = not get_option('disable-threading') and get_option('enable-openmp') and cpp.has_argument('-fopenmp')
+summary({
+  'Build with openMP? ' : use_openmp,
+})
+
+# Setup openmp flags for x86-simd-sort:
+omp_cflags = []
+omp = []
+if use_intel_sort and use_openmp
+  omp = dependency('openmp', required : false)
+  if omp.found()
+    omp_cflags = ['-fopenmp', '-DXSS_USE_OPENMP']
```

Let's not have a separate variable for this. It's more idiomatic to use a dependency object:

```suggestion
omp_dep = declare_dependency(dependencies: omp, compile_args: ['-DXSS_USE_OPENMP'])
```

The `-fopemp` flag shouldn't need to be added explicitly, it's already present in the `omp` dependency (e.g., see [here](https://github.com/mesonbuild/meson/blob/59d3c427f53e87df2d10e8e314f1789fc9d6ddfb/mesonbuild/compilers/mixins/intel.py#L85-L89)).

---

Sounds good to me. I have also updated the dependency `required` flag to be true. If someone is explicitly using  `enable-openmp=true`, then its reasonable to expect a build failure if openMP isn't available.